catroid/src/test/java/org/catrobat/catroid/test/stage/ShowBubbleActorTest.java

```diff
@@ -0,0 +1,170 @@
+public class ShowBubbleActorTest {
+	private Image image;
+	private Sprite sprite;
+	private static final String OBJECT_NAME = "Test";
+	private static final Integer CALC_X_LEFT = -347;
+	private static final Integer CALC_Y_LEFT = 347;
+	private static final Integer CALC_X_RIGHT = 347;
+	private static final Integer CALC_Y_RIGHT = 347;
+	private static final Float IMAGE_WIDTH = 148.0f;
+	private static final Float EXPECTED_Y_STD = 347.0f;
+	private static final Float EXPECTED_Y_MOVE_Y = -153.0f;
+	private static final Float EXPECTED_X_STD_RIGHT = 347.0f;
+	private static final Float EXPECTED_X_STD_LEFT = -495.0f;
+	private static final Float SPRITE_WIDTH_DEFAULT = 1080.0f;
+	private static final Float SPRITE_HEIGHT_DEFAULT = 1794.0f;
+	private static final Float SPRITE_Y_DEFAULT = -897.0f;
+	private static final Float SPRITE_X_DEFAULT = -540.0f;
+	private static final Float SPRITE_X_FOR_MOVE_X = -40.0f;
+	private static final Float SPRITE_Y_FOR_MOVE_Y = -1397.0f;
+
+	@Before
+	public void setUp() throws Exception {
+		image = PowerMockito.mock(Image.class);
+		Pixmap pixmap = PowerMockito.mock(Pixmap.class);
+		Bitmap bitmap = PowerMockito.mock(Bitmap.class);
+		Texture texture = PowerMockito.mock(Texture.class);
+
+		sprite = new Sprite(OBJECT_NAME);
+		sprite.look.setX(SPRITE_X_DEFAULT);
+		sprite.look.setY(SPRITE_Y_DEFAULT);
+		sprite.look.setHeight(SPRITE_HEIGHT_DEFAULT);
+		sprite.look.setWidth(SPRITE_WIDTH_DEFAULT);
+
+		Mockito.when(image.getWidth()).thenReturn(IMAGE_WIDTH);
+
+		PowerMockito.mockStatic(Bitmap.class);
+
+		Mockito.when(Bitmap.createBitmap(Mockito.any(Integer.class), Mockito.any(Integer.class),
+				Mockito.any(Bitmap.Config.class))).thenReturn(bitmap);
+
+		PowerMockito.whenNew(Image.class).withParameterTypes(Texture.class)
+				.withArguments(Mockito.any()).thenReturn(image);
+
+		PowerMockito.whenNew(Pixmap.class).withArguments(Mockito.anyByte(), Mockito.anyInt(),
+				Mockito.anyInt()).thenReturn(pixmap);
+
+		PowerMockito.whenNew(Texture.class).withParameterTypes(Pixmap.class)
+				.withArguments(Mockito.any()).thenReturn(texture);
+	}
+
+	@Test
+	public void testLeftBubblePosition() {
+		Float expectedXLeft = EXPECTED_X_STD_LEFT;
+		Float expectedYLeft = EXPECTED_Y_STD;
+
+		ShowBubbleActor bubbleActor = new ShowBubbleActor(OBJECT_NAME, sprite, Constants.SAY_BRICK);
+
+		assertEquals(expectedXLeft, bubbleActor.calculateLeftImageX(CALC_X_LEFT));
+		assertEquals(expectedYLeft, bubbleActor.calculateImageY(CALC_Y_LEFT));
+	}
+
+	@Test
+	public void testRightBubblePosition() {
+		Float expectedXRight = EXPECTED_X_STD_RIGHT;
+		Float expectedYRight = EXPECTED_Y_STD;
+
+		ShowBubbleActor bubbleActor = new ShowBubbleActor(OBJECT_NAME, sprite, Constants.SAY_BRICK);
+
+		assertEquals(expectedXRight, bubbleActor.calculateRightImageX(CALC_X_RIGHT));
+		assertEquals(expectedYRight, bubbleActor.calculateImageY(CALC_Y_RIGHT));
+	}
+
+	@Test
+	public void testLeftBubblePositionAfterMoveX() {
+		Float expectedXLeft = 5.0f;
```

Comment:
Is there something special about this value? Else could you also declare this variable as a constant.